service/raft/raft_group0.hh

```diff
@@ -110,11 +101,31 @@ public:
     // it can't do it by itself (it's dead).
     future<> leave_group0(std::optional<gms::inet_address> host = {});
 
+private:
     void init_rpc_verbs();
     future<> uninit_rpc_verbs();
 
     // Handle peer_exchange RPC
     future<group0_peer_exchange> peer_exchange(discovery::peer_list peers);
+
+    raft_server_for_group create_server_for_group(raft::group_id id, raft::server_address my_addr);
+
+    // Loads server address for group 0 from disk if present, otherwise randomly generates a new one and persists it.
+    // Execute on shard 0 only.
+    future<raft::server_address> load_or_create_my_addr();
+
+    // Run the discovery algorithm.
+    //
+    // Discovers an existing group 0 cluster or elects a server (called a 'leader')
+    // responsible for creating a new group 0 cluster if one doesn't exist
+    // (in particular, we may become that leader).
+    //
+    // See 'raft-in-scylla.md', 'Establishing group 0 in a fresh cluster'.
+    future<group0_info> discover_group0(raft::server_address my_addr);
+
+    // Called by `discover_group0`.
+    // Precondition: `_group0` contains `persistent_discovery`.
```

Comment:
What the heck is `persistent_discovery`

Reply:
https://github.com/scylladb/scylla/blob/master/service/raft/raft_group0.hh#L22

Reply:
I can grep it myself, but if we have a comment why not make it more informative and explain in words what state the protocol should be to use the function. Though this is pre-existing problem.

Reply:
I remove `do_discover_group0` in a later commit, replacing it with a static free-function in raft_group0.cc. So that comment is gone